packages/ui/src/domainModules/collectionMammals/components/MammalManager/MainColumn/RecordForm/formParts/RecordHistoryEvents/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { Grid, Header } from 'semantic-ui-react'
+
+import { withI18n } from 'coreModules/i18n/higherOrderComponents'
+
+import EventRow from './EventRow'
+
+const mapStateToProps = (state, { formValueSelector, name }) => {
+  return {
+    recordHistoryEvents: formValueSelector(state, name),
+  }
+}
+
+const propTypes = {
+  i18n: PropTypes.shape({
+    moduleTranslate: PropTypes.func.isRequired,
+  }).isRequired,
+  recordHistoryEvents: PropTypes.arrayOf(
+    PropTypes.shape({
+      action: PropTypes.string,
+      srcCreatedAt: PropTypes.string,
+      srcUpdatedAt: PropTypes.string,
+      userName: PropTypes.string,
+    })
+  ),
+}
+const defaultProps = {
+  recordHistoryEvents: [],
+}
+
+class RecordHistoryEvents extends Component {
+  render() {
+    const { i18n: { moduleTranslate }, recordHistoryEvents } = this.props
+
+    const createdEvent = recordHistoryEvents.find(
+      ({ action }) => action === 'create'
+    )
+    const lastModifiedEvent = recordHistoryEvents.find(
+      ({ action }) => action === 'update'
+    )
+
+    const hasEvents = recordHistoryEvents.length > 0
```

Comment:
those three, extract them in mapStateToProps instead and do not send recordHistoryEvents to the component. that way the component will only rerender if any of those has changed. it will not rerender due to other changes in recordHistoryEvents. then this should also be possible to have as a PureComponent